devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy

```diff
@@ -11,100 +11,136 @@
-		return result
-	}
-}
+    tiles(scale: 2) {
+        multiAttributeTile(name:"valve", type: "generic", width: 6, height: 4, canChangeIcon: true){
+            tileAttribute ("device.contact", key: "PRIMARY_CONTROL") {
+                attributeState "open", label: '${name}', action: "valve.close", icon: "st.valves.water.open", backgroundColor: "#53a7c0", nextState:"closing"
+                attributeState "closed", label: '${name}', action: "valve.open", icon: "st.valves.water.closed", backgroundColor: "#e86d13", nextState:"opening"
+                attributeState "opening", label: '${name}', action: "valve.close", icon: "st.valves.water.open", backgroundColor: "#53a7c0", nextState:"closing"
+                attributeState "closing", label: '${name}', action: "valve.open", icon: "st.valves.water.closed", backgroundColor: "#e86d13", nextState:"opening"
+            }
+            tileAttribute ("powerType", key: "SECONDARY_CONTROL") {
+                attributeState "powerType", label:'Power: ${currentValue}'
+            }
+        }
+
+        valueTile("battery", "device.battery", inactiveLabel:false, decoration:"flat", width:2, height:2) {
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
 
-// Commands to device
-def on() {
-	log.debug "on()"
-	sendEvent(name: "switch", value: "on")
-	"st cmd 0x${device.deviceNetworkId} 1 6 1 {}"
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+
+        main(["valve"])
+        details(["valve", "battery", "refresh"])
+    }
 }
 
-def off() {
-	log.debug "off()"
-	sendEvent(name: "switch", value: "off")
-	"st cmd 0x${device.deviceNetworkId} 1 6 0 {}"
+private getCLUSTER_BASIC() { 0x0000 }
+private getBASIC_ATTR_POWER_SOURCE() { 0x0007 }
+private getCLUSTER_POWER() { 0x0001 }
+private getPOWER_ATTR_BATTERY_PERCENTAGE_REMAINING() { 0x0021 }
+private getTYPE_U8() { 0x20 }
+private getTYPE_ENUM8() { 0x30 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        if(event.name == "switch") {
+            event.name = "contact"			//0006 cluster in valve is tied to contact
+            if(event.value == "on") {
+                event.value = "open"
+            }
+            else if(event.value == "off") {
+                event.value = "closed"
+            }
+            sendEvent(event)
+        }
+        else {
+            sendEvent(event)
+        }
+    }
+    else {
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        if (descMap?.clusterInt == 0 && descMap?.attrInt == 7){
+            def value = descMap?.value
+            if (value == "01" || value == "02") {
+                sendEvent(name: "powerType", value: "Mains")
```

Comment:
only problem is every time the device says, i am still on battery, this will send out a notifcation. will remove this. it should be done from the smartapp. we'll need a capability for the powerType